### PR TITLE
Address UG and DG changes and fix find and sort diagrams

### PR DESCRIPTION
**Pull request by Nanosync**
- Addresses UG and DG changes from review on 24/10
- Fixed find and sort diagrams
- Added preface for _some_ sections and description of diagrams

#89

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -86,44 +86,49 @@ Format: `help`
...
 
 [NOTE]
-You do not need to specify which date format you would like to use as the application will be able to differentiate it from the input received.
+You do not need to specify which date format you would like to use as MoneyGoWhere will be able to differentiate it from the input received.
 
 [NOTE]
-The application will also automatically check whether a given date is valid.
-Should you accidentally provide an invalid date such as 31/02/2019, it will be rejected as there are only 28 days in February 2019.
+MoneyGoWhere will also automatically check whether a given date is valid.
+Should you accidentally provide an invalid date such as `31/02/2019`, it will be rejected as there are only 28 days in February 2019.
```
Comment (jonathantjendana) [inline docs/UserGuide.adoc]: `31/02/2019` should not be highlighted as this format is reserved for command key words as specified in section 2, How to use this guide.

**Review by austinsantoso — APPROVED** [review]
lgtm

**Review by jonathantjendana — CHANGES_REQUESTED** [review]

File: docs/UserGuide.adoc (hunk 1 above)
Comment (Nanosync) [reply docs/UserGuide.adoc]: Changed!

**Review by jonathantjendana — APPROVED** [review]
lgtm!

**Pull request merged**